src/illuminatio/test_orchestrator.py

```diff
@@ -36,90 +46,113 @@ def getManifest(yaml_filename):
+            # choose any port for wildcard
+            rewritten_ports[wildcard_port] = prefix + str(service_ports[0].port)
+        for port in numbered_ports:
+            prefix = "-" if "-" in port else ""
+            port_int = int(port.replace("-", ""))
+            service_ports_for_port = [p for p in service_ports if p.target_port == port_int]
+            # TODO this was a hotfix for recipe 11, where ports 53 were allowed but not for any target,
+            # resulting in test to 53 being written despite no service matching them existing.
+            # That error should be handled in test generation, an exception here would be fine
+            if service_ports_for_port:
+                rewritten_ports[port] = prefix + str(service_ports_for_port[0].port)
+            else:
+                # TODO change to exception, handle it higher up
+                rewritten_ports[port] = "err"
+        return rewritten_ports
 
     def _get_target_names_creating_them_if_missing(self, target_dict, api: k8s.client.CoreV1Api):
         svc_names_per_host = {}
         port_dict_per_host = {}
         for host_string in target_dict.keys():
             host = Host.from_identifier(host_string)
             if isinstance(host, GenericClusterHost):
-                logger.debug("Found GenericClusterHost " + str(
+                self.logger.debug("Found GenericClusterHost " + str(
                     host) + ". Rewriting it to a ClusterHost in default namespace now.")
                 host = ClusterHost("default", host.pod_labels)
             if not isinstance(host, ClusterHost):
                 raise ValueError("Only ClusterHost targets are supported by this Orchestrator. Host: " + str(
                     host) + ", hostString: " + host_string)
-            logger.debug("Searching service for host " + str(host))
+            self.logger.debug("Searching service for host " + str(host))
             services_for_host = [svc for svc in self._current_services if host.matches(svc)]
-            logger.debug("Found services {} for host {} ".format([svc.metadata for svc in services_for_host], host))
+            self.logger.debug("Found services {} for host {} ".format([svc.metadata for svc in services_for_host],
+                                                                      host))
             rewritten_ports = self._rewrite_ports_for_host(target_dict[host_string], services_for_host)
-            logger.debug("Rewritten ports: " + str(rewritten_ports))
+            self.logger.debug("Rewritten ports: " + str(rewritten_ports))
             port_dict_per_host[host_string] = rewritten_ports
             if not services_for_host:
                 gen_name = PROJECT_PREFIX + "-test-target-pod-"
-                target_container = k8s.client.V1Container(image=self.target_image, name="runner")
+                target_container = k8s.client.V1Container(image=self.docker_image["target"], name="runner")
```

Comment:
Don't use the name `docker` in the code please (if not appropriate because we actually interact with the docker daemon.